### PR TITLE
Remove session contract indirection when finding identity of actor

### PR DESCRIPTION
**Pull request by LucianBuzzo**
With this change, autumndb sessions are now expected to be objects
containing the actor and an optional scope directly instead of a string
referring to the session contract. Validating session contracts
should now be handled by the implementing app layer.
This means that autumndb is only dealing with identity, instead of
partially handling auth.

Change-type: major
Signed-off-by: Lucian Buzzo <lucian.buzzo@gmail.com>

### CODE REVIEW

**Comment by LucianBuzzo** [comment]
@balena-ci retest

**Comment by LucianBuzzo** [comment]
@balena-ci retest

**Comment by LucianBuzzo** [comment]
@balena-ci retest

**Comment by LucianBuzzo** [comment]
Testing this upstream [hear](https://github.com/product-os/jellyfish/pull/8749).

**Comment by LucianBuzzo** [comment]
@Ereski Everything is passing in the upstream draft PRs, can you take a look at this? When [I tested it on JF](https://github.com/product-os/jellyfish/pull/8749) I saw a ~20% performance boost on contract creation, plus we now have a clear path to switching to a 3rd party for authorization.

File: test/integration/contracts/role-user-community.spec.ts [hunk 1]
```diff
@@ -1,16 +1,14 @@
-import { ViewContract } from '@balena/jellyfish-types/build/core';
+import { UserContract, ViewContract } from '@balena/jellyfish-types/build/core';
```
Comment (joshbwlng) [inline test/integration/contracts/role-user-community.spec.ts:1]:
```suggestion
import type { UserContract, ViewContract } from '@balena/jellyfish-types/build/core';
```

File: lib/kernel.ts [hunk 2]
```diff
@@ -457,7 +459,7 @@ export class Kernel {
 		]);
 
 		const adminUser = await unsafeUpsert(CONTRACTS['user-admin']);
-		const adminSession = await unsafeUpsert({
+		await unsafeUpsert({
```
Comment (Ereski) [inline lib/kernel.ts:462]: shouldn't this be removed entirely?

**Review by Ereski — APPROVED** [review]

**Comment by Ereski** [comment]
@LucianBuzzo just a note

> When https://github.com/product-os/jellyfish/pull/8749 I saw a ~20% performance boost on contract creation

JF does a LOT of back and forth between session IDs, session contracts, user IDs and user contracts. I'm sure that just rationalizing how these are used can remove quite a few unnecessary queries.

File: lib/kernel.ts (hunk 2 above)
Comment (joshbwlng) [reply lib/kernel.ts]: @Ereski Nice catch, removed it and confirmed that everything passes. Rebased and pushed the changes.

**Comment by LucianBuzzo** [comment]
@balena-ci retest

**Pull request merged**